### PR TITLE
fix: generify WebsocketRouteOptions

### LINKED ISSUE #127: Types: WebsocketRouteOptions is not generic and results in handler not being typed properly
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure it has not already been reported

### Fastify version

3.15.1

### Node.js version

14.5.0

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

20.04

### Description

The `WebsocketRouteOptions` doesn't accept generics, so `WebsocketHandler` is left with the defaults, not with the appropriate types.

### Steps to Reproduce

```ts
import fastify from "fastify"
import fastifyWs from "fastify-websocket"

const fastify = fastify()
fastify.register(fastifyWs)
```
Let's say we have this interface declared:
```ts
interface AuthenticatedEndpoint extends RequestGenericInterface {
  Body: {
    auth: string
  }
}
```
When using it with `app.route`:
```ts
fastify.route<AuthenticatedEndpoint>({
  method: 'GET',
  url: '/ws',
  wsHandler(conn, req)  {
    req.body // typed as `unknown`
    req.body.auth // TypeError
  }
})
```

### Expected Behavior

(assuming we have the same starting code as we had in Steps to Reproduce)

```ts
fastify.route<AuthenticatedEndpoint>({
  method: 'GET',
  url: '/ws',
  wsHandler(conn, req)  {
    req.body // typed as { auth: string }
    req.body.auth // string
  }
})

### PR DESCRIPTION
**Pull request by VottusCode**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

fixes #127

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,45 +1,76 @@
...
-  wsHandler?: WebsocketHandler
+interface WebsocketRouteOptions<
+  RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestDefaultExpression<RawServer> = RawRequestDefaultExpression<RawServer>,
+  RequestGeneric extends RequestGenericInterface = RequestGenericInterface
+> {
+  wsHandler?: WebsocketHandler<RawServer, RawRequest, RequestGeneric>;
 }
-declare module 'fastify' {
+declare module "fastify" {
```
Comment (VottusCode) [inline index.d.ts:27]: I swear to god Prettier is gonna kill me one day

Comment (Fdawgs) [reply index.d.ts:27]: > 
> 
> I swear to god Prettier is gonna kill me one day

If you can revert these style changes then i'll approve a workflow run, or else `standard` is going to throw a fit during the CI tests.

Comment (VottusCode) [reply index.d.ts:27]: Okay

File: test/types/index.test-d.ts [hunk 2]
```diff
@@ -65,7 +66,7 @@ const augmentedRouteOptions: RouteOptions = {
   },
   wsHandler: (connection, request) => {
     expectType<SocketStream>(connection);
-    expectType<FastifyRequest<RequestGenericInterface>>(request)
+    expectType<FastifyRequest<RouteGenericInterface>>(request)
   },
 };
```
Comment (airhorns) [inline test/types/index.test-d.ts:71]: Want to add a test for the new functionality based on your original issue? Thanks for the contribution!

Comment (VottusCode) [reply test/types/index.test-d.ts:71]: I never really tested "types", didn't know that was a thing until now :D

Comment (VottusCode) [reply test/types/index.test-d.ts:71]: So... what next then? :D

**Comment by VottusCode** [comment]
Can this be merged?

File: test/types/index.test-d.ts (hunk 2 above)
Comment (airhorns) [reply test/types/index.test-d.ts:71]: To add a test case you'd need to copy one of these stanzas and exercise specifically the new types you added! This file is using `tsd` which is a typescript testing tool and you can find more info in its docs.

**Comment by mcollina** [comment]
I'm sorry, CI was stuck @fastify/typescript could you take a look?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by airhorns — CHANGES_REQUESTED** [review]
Ready to go after adding a test!

**Comment by kibertoad** [comment]
@VottusCode Do you need any help with this :)?

**Comment by VottusCode** [comment]
> @VottusCode Do you need any help with this :)?
@kibertoad I haven't had time at all due to work combined with other IRL stuff, I will take a look at it.

**Comment by airhorns** [comment]
Merged in #176

**Pull request closed without merging**